ci/concourse/pipelines/kf-pipeline.yml

```diff
@@ -217,3 +235,99 @@ jobs:
             ./fly -t "$FLY_TARGET" destroy-pipeline -n -p $extra_pipeline
           done
           rm -f $prs $pipelines $extra
+
+# nightly build of kf artifacts
+- name: nightly-build
+  plan:
+  - aggregate:
+    - get: kf-master
+      trigger: false
+    - get: nightly-build-trigger
+      trigger: true
+  - task: ko-resolve
+    config:
+      platform: linux
+      image_resource:
+        type: docker-image
+        source: *ci-image-source
+      params:
+        SERVICE_ACCOUNT_JSON: *release_service_account_json
+        GCP_PROJECT_ID: *gcp_project_id
+        KO_DOCKER_REPO: *ko_docker_release_repo
+        RELEASE_BUCKET: *release_bucket
+        CLI_RELEASE_BUCKET: *cli_release_bucket
+      inputs:
+      - name: kf-master
+      run:
+        dir: kf-master
+        path: sh
+        args:
+        - -exc
+        - |
+
+          # Login to gcloud
+          set +x
+          /bin/echo "$SERVICE_ACCOUNT_JSON" > key.json
+          set -x
+          /bin/echo Authenticating to kubernetes...
+          gcloud auth activate-service-account --key-file key.json
+          gcloud config set project "$GCP_PROJECT_ID"
+          gcloud -q auth configure-docker
+
+          # Used to suffix the artifacts
+          current_time=$(date +%s)
+
+          #########################
+          # Generate Release YAML #
+          #########################
+
+          # Environment Variables for go build
+          export GOPATH=/go
+          export GOPROXY=https://proxy.golang.org
+          export GOSUMDB=sum.golang.org
+          export GO111MODULE=on
+          export CGO_ENABLED=0
+
+          # ko requires a proper go path and deps to be vendored
+          # TODO remove this once https://github.com/google/ko/issues/7 is
+          # resolved.
+          go mod vendor
+          mkdir -p $GOPATH/src/github.com/google/
+          ln -s $PWD $GOPATH/src/github.com/google/kf
+          cd $GOPATH/src/github.com/google/kf
+
+          release_name=release-${current_time}.yaml
+          # ko resolve
+          # This publishes the images to KO_DOCKER_REPO and writes the yaml to
+          # stdout.
+          ko resolve --filename config > /tmp/${release_name}
+
+          # save release to a bucket
+          gsutil cp /tmp/${release_name} ${RELEASE_BUCKET}/${release_name}
```

Comment:
Should these file transfers be encapsulated in a Concourse `put`? Just wondering what happens if this succeeds, but the `go build` that follows fails.

Reply:
This is an interesting idea. However if we move this to a hack script (so we can run it manually if we need to), then I'm not sure the concourse put works as well. WDYT?

Reply:
Definitely doesn't work with the hack script. And I like the hack scripts a lot. LGTM!